File: leno/src/main/java/hello/KotlinHello.java
package hello;

import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import poc.TestBean;


public class KotlinHello {
    public static void main(String[] args){
        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
        context.register(TestCenter.class);
        ConfigurableListableBeanFactory beanFactory = context.getBeanFactory();
        TestBean testBean = new TestBean();
        beanFactory.registerSingleton("TestBean", testBean);
        //beanFactory.registerSingleton("HelloBean", HelloBean(testBean))
        context.refresh();
        //println(context.getBean("HelloBean"))
        //HelloBean().say()
        ((TestBean) context.getBean("TestBean")).setMessage("Hello Bean55");
        ((HelloBean) context.getBean("HelloBean")).print();
        new HelloBean().print();
        context.close();
    }
}

File: leno/src/main/kotlin/hello/KotlinHello.kt
package hello

import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.AnnotationConfigApplicationContext
import poc.TestBean

val KotlinHelloString : String = "Hello from Kotlin!"

fun getHelloStringFromJava() : String {
    return JavaHello.JavaHelloString!!;
}

fun getHelloString() : String {
    return "Hello, world!"
}

fun main(args : Array<String>) {
    //println(getHelloString())
    //print(getHelloStringFromJava())
    var context = AnnotationConfigApplicationContext()
    context.register(TestCenter::class.java)
    var beanFactory = context.beanFactory
    val testBean = TestBean()
    beanFactory.registerSingleton("TestBean", testBean)
    //beanFactory.registerSingleton("HelloBean", HelloBean(testBean))
    context.refresh()
    //println(context.getBean("HelloBean"))
    //HelloBean().say()
    //(context.getBean("TestBean") as TestBean).setMessage("Hello Bean55")
    //(context.getBean("HelloBean") as HelloBean).print()
    HelloBean().print()
    context.close()
}
